.github/workflows/pr.yaml

```diff
@@ -37,15 +37,20 @@ jobs:
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
         with:
           fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
 
       - name: Check Image and Relevant Changes
         id: check-image
         uses: ./.github/actions/check-image-and-changes
 
       - name: Check if build should be skipped
         run: |
+          if [[ "${{ steps.check-image.outputs.is_skipped }}" == "true" ]]; then
+            echo "::notice::Skipping build: [skip-build] tag found in commit messages."
+            exit 0
```

Comment:
as discussed in the other PR, exit 0 does NOT exit from the workflow ... so you need to add all the other `if:` checks below here if you want to potentially stop doing things from this point on.